matsim/src/main/java/ch/sbb/matsim/routing/pt/raptor/SwissRailRaptorData.java

```diff
@@ -42,547 +42,603 @@
-        final int toRouteStop;
-        final int transferTime;
-        final int transferDistance;
-
-        RTransfer(int fromRouteStop, int toRouteStop, double transferTime, double transferDistance) {
-            this.fromRouteStop = fromRouteStop;
-            this.toRouteStop = toRouteStop;
-            this.transferTime = (int) Math.ceil(transferTime);
-            this.transferDistance = (int) Math.ceil(transferDistance);
-        }
-    }
-    
-    /*
-     * synchronized in order to avoid that multiple quad trees for the very same stop filter attribute/value combination are prepared at the same time 
-     */
-	public synchronized void prepareStopFilterQuadTreeIfNotExistent(String stopFilterAttribute, String stopFilterValue) {
+	private static final Logger log = Logger.getLogger(SwissRailRaptorData.class);
+
+	final RaptorStaticConfig config;
+	final int countStops;
+	final int countRouteStops;
+	final RRoute[] routes;
+	final int[] departures; // in the RAPTOR paper, this is usually called "trips", but I stick with the
+							// MATSim nomenclature
+	final Vehicle[] departureVehicles; // the vehicle used for each departure
+	final Id<Departure>[] departureIds;
+	final RRouteStop[] routeStops; // list of all route stops
+	final RTransfer[] transfers;
+	final Map<TransitStopFacility, Integer> stopFacilityIndices;
+	final Map<TransitStopFacility, int[]> routeStopsPerStopFacility;
+	final QuadTree<TransitStopFacility> stopsQT;
+	final Map<String, Map<String, QuadTree<TransitStopFacility>>> stopFilterAttribute2Value2StopsQT;
+	final OccupancyData occupancyData;
+
+	private SwissRailRaptorData(RaptorStaticConfig config, int countStops, RRoute[] routes, int[] departures,
+			Vehicle[] departureVehicles, Id<Departure>[] departureIds, RRouteStop[] routeStops, RTransfer[] transfers,
+			Map<TransitStopFacility, Integer> stopFacilityIndices,
+			Map<TransitStopFacility, int[]> routeStopsPerStopFacility, QuadTree<TransitStopFacility> stopsQT,
+			OccupancyData occupancyData) {
+		this.config = config;
+		this.countStops = countStops;
+		this.countRouteStops = routeStops.length;
+		this.routes = routes;
+		this.departures = departures;
+		this.departureVehicles = departureVehicles;
+		this.departureIds = departureIds;
+		this.routeStops = routeStops;
+		this.transfers = transfers;
+		this.stopFacilityIndices = stopFacilityIndices;
+		this.routeStopsPerStopFacility = routeStopsPerStopFacility;
+		this.stopsQT = stopsQT;
+		this.stopFilterAttribute2Value2StopsQT = new HashMap<>();
+		this.occupancyData = occupancyData;
+	}
+
+	public static SwissRailRaptorData create(TransitSchedule schedule, @Nullable Vehicles transitVehicles,
+			RaptorStaticConfig staticConfig, Network network, OccupancyData occupancyData) {
+		log.info("Preparing data for SwissRailRaptor...");
+		long startMillis = System.currentTimeMillis();
+
+		Map<Id<Vehicle>, Vehicle> vehicles = transitVehicles == null ? Collections.emptyMap()
+				: transitVehicles.getVehicles();
```

Comment:
please set a wider line limit, this is actually much more difficult to read than when it's all on one line